File: src/main/java/org/flossware/civilization/model/PoliticsState.java
package org.flossware.civilization.model;

/**
 * Immutable political state snapshot.
 */
public record PoliticsState(
    double stability,
    String government,
    int rulerAge,
    double warExhaustion,
    boolean inRebellion,
    boolean inSuccessionCrisis
) {
    public PoliticsState {
        if (stability < 0 || stability > 1) {
            throw new IllegalArgumentException("Stability must be in [0, 1]");
        }
        if (rulerAge < 0) {
            throw new IllegalArgumentException("Ruler age cannot be negative");
        }
        if (warExhaustion < 0 || warExhaustion > 1) {
            throw new IllegalArgumentException("War exhaustion must be in [0, 1]");
        }
    }

    public PoliticsState withStability(double newStability) {
        double clamped = Math.max(0, Math.min(1, newStability));
        return new PoliticsState(clamped, government, rulerAge, warExhaustion, inRebellion, inSuccessionCrisis);
    }

    public PoliticsState withRebellion(boolean isInRebellion) {
        return new PoliticsState(stability, government, rulerAge, warExhaustion, isInRebellion, inSuccessionCrisis);
    }

    public PoliticsState withSuccessionCrisis(boolean isCrisis) {
        return new PoliticsState(stability, government, rulerAge, warExhaustion, inRebellion, isCrisis);
    }

    public PoliticsState withWarExhaustion(double newExhaustion) {
        double clamped = Math.max(0, Math.min(1, newExhaustion));
        return new PoliticsState(stability, government, rulerAge, clamped, inRebellion, inSuccessionCrisis);
    }

    public PoliticsState withRulerAge(int newAge) {
        return new PoliticsState(stability, government, newAge, warExhaustion, inRebellion, inSuccessionCrisis);
    }
}

File: mobile/shared/src/commonMain/kotlin/org/flossware/civilization/model/PoliticsState.kt
package org.flossware.civilization.model

import kotlinx.serialization.Serializable

@Serializable
data class PoliticsState(
    val stability: Double,
    val government: String,
    val rulerAge: Int,
    val warExhaustion: Double,
    val inRebellion: Boolean,
    val inSuccessionCrisis: Boolean
) {
    init {
        require(stability in 0.0..1.0) { "Stability must be in [0, 1]" }
        require(rulerAge >= 0) { "Ruler age cannot be negative" }
        require(warExhaustion in 0.0..1.0) { "War exhaustion must be in [0, 1]" }
    }

    fun withStability(newStability: Double) = copy(stability = newStability.coerceIn(0.0, 1.0))
    fun withRebellion(isInRebellion: Boolean) = copy(inRebellion = isInRebellion)
    fun withSuccessionCrisis(isCrisis: Boolean) = copy(inSuccessionCrisis = isCrisis)
    fun withWarExhaustion(newExhaustion: Double) = copy(warExhaustion = newExhaustion.coerceIn(0.0, 1.0))
    fun withRulerAge(newAge: Int) = copy(rulerAge = newAge)
}
